NotebookAnalyzers/entities.py

```diff
@@ -48,8 +73,6 @@ class GitHubRepository(Repository):
     """
 
     def __init__(self, github_url: str):
-        # Clones the repo
-        # Crawls the repo's content searching for notebooks
-        # When it finds a notebook, it appends it to the list of notebooks
-        # e.g., notebooks.append(Notebook(path_where_the_nb_was_found))
-        ...
+        git.Git(config.data_path).clone(github_url)
```

Comment:
I think a good idea here would be limiting the clone operation to just the last version of the repo.

It should be as easy as adding the `depth=1`  argument to the function call; [here](https://github.com/gitpython-developers/GitPython/issues/269#issuecomment-77743473) is an example.